### PR TITLE
Integrate components to the loop.

### PR DESCRIPTION
**Pull request by Eleonore9**
**Aim**: integrate the fertility, mortality and migration modules into the projection loop.

**What's new**:
- Higher level workflow functions in the fertility, mortality and migration namespaces to link the lower level workflow functions together.
- Integrate those new workflow functions inside the `looping-test` in `projection_loop.clj`.
- Update the test namespaces so that the handmade data inputs are replaced by real inputs (from other modules).
- Update of the schemas and keywords to harmonise across namespaces.
- Data output from R to run the projection loop until 2040 and compare Clojure results with R results (in `projection_loop_test.clj`)
- Remove the "District" column from data and code as it is not used. We should later provide a way to look up districts from Gss codes.
- Rename `:pm` to `:proportion-male-newborns`. 

**Note for the "Worspace team"**:
Feedback welcome on how we combined workflow functions and used `merge->`!
For our feedback see https://www.pivotaltracker.com/projects/1610511/stories/123122401

### CODE REVIEW

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 1]
```diff
@@ -170,3 +170,37 @@
...
+
+(defworkflowfn births-projection
+  "Handles the steps of the fertility module happening inside of the loop.
+   Takes in the projected ASFR and the population at risk from the loop.
+   Returns the number of births by sex for projection year."
+  {:witan/name :ccm-fert/births-projection
+   :witan/version "1.0"
+   :witan/input-schema {:initial-projected-fertility-rates ProjFixedASFRSchema
+                        :population-at-risk PopulationSchema}
+   :witan/param-schema {:pm double}
```
Comment (acron0) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: `:pm` is an ambiguous key name

File: test/witan/models/dem/ccm/core/projection_loop_test.clj [hunk 2]
```diff
@@ -4,24 +4,56 @@
...
+                   :domestic-in-migrants
+                   "test_data/model_inputs/mig/bristol_hist_domestic_inmigrants.csv"
+                   :domestic-out-migrants
+                   "test_data/model_inputs/mig/bristol_hist_domestic_outmigrants.csv"
+                   :international-in-migrants
+                   "test_data/model_inputs/mig/bristol_hist_international_inmigrants.csv"
+                   :international-out-migrants
+                   "test_data/model_inputs/mig/bristol_hist_international_outmigrants.csv"}))
+
+(def params {:fert-last-yr 2014 :pm (double (/ 105 205))
```
Comment (acron0) [inline test/witan/models/dem/ccm/core/projection_loop_test.clj]: I find the formatting of these params pretty hard to read. clj-fmt perhaps?

Comment (Eleonore9) [inline test/witan/models/dem/ccm/core/projection_loop_test.clj]: Cljfmt doesn't find any issue, but I'll try and make it easier to read.

**Comment by acron0** [comment]
Thanks :shipit:

**Comment by seb231** [comment]
If it works, it lgtm! :) :shipit:

**Pull request merged**